src/components/Arrow/Arrow.test.tsx

```diff
@@ -1,47 +1,173 @@
-import { render } from "@testing-library/react";
+import { fireEvent, render, screen } from "@testing-library/react";
 import * as React from "react";
-import { Arrow, ArrowType } from "./Arrow";
+import { Arrow } from "./Arrow";
+import { ArrowType } from "./Utils";
 
 describe(Arrow.name, () => {
-  it("should render", () => {
+  it("should have a body and an arrow head if the arrow is directional.", () => {
     const arrow = render(
       <Arrow
         start={{ x: 0, y: 0 }}
         end={{ x: 100, y: 0 }}
-        color="#000000"
+        arrowColor="#000000"
+        circleColor="#FFFFFF"
+        iconColor="#FFFFFF"
         type={ArrowType.Directional}
+        notes=""
+        completed={false}
       />,
     ).container;
 
     expect(arrow.querySelector("div")).toBeTruthy();
     expect(arrow.querySelectorAll("svg").length).toBe(2);
   });
 
-  it("should add two arrow heads if the arrow is bidirectional", () => {
+  it("should add two arrow heads if the arrow is bidirectional.", () => {
     const arrow = render(
       <Arrow
         start={{ x: 0, y: 0 }}
         end={{ x: 100, y: 0 }}
-        color="#000000"
+        arrowColor="#000000"
+        circleColor="#FFFFFF"
+        iconColor="#FFFFFF"
         type={ArrowType.BiDirectional}
+        notes=""
+        completed={false}
       />,
     ).container;
 
     expect(arrow.querySelector("div")).toBeTruthy();
     expect(arrow.querySelectorAll("svg").length).toBe(3);
   });
 
-  it("should render", () => {
+  it("should only have one svg, the arrow body, if the arrow is nondirectional.", () => {
     const arrow = render(
       <Arrow
         start={{ x: 0, y: 0 }}
         end={{ x: 100, y: 0 }}
-        color="#000000"
+        arrowColor="#000000"
+        circleColor="#FFFFFF"
+        iconColor="#FFFFFF"
         type={ArrowType.NonDirectional}
+        notes=""
+        completed={false}
       />,
     ).container;
 
     expect(arrow.querySelector("div")).toBeTruthy();
     expect(arrow.querySelectorAll("svg").length).toBe(1);
   });
+
+  it("nondirectional arrow should have icon when hovered over", () => {
+    const arrow = render(
+      <div data-testid="arrowTest">
+        <Arrow
+          start={{ x: 0, y: 0 }}
+          end={{ x: 100, y: 0 }}
+          arrowColor="#000000"
+          circleColor="#FFFFFF"
+          iconColor="#FFFFFF"
+          type={ArrowType.NonDirectional}
+          notes=""
+          completed={false}
+        />
+      </div>,
```

Comment:
Do we use `div[data-testid="arrowTest"]`?

```suggestion
      <Arrow
        start={{ x: 0, y: 0 }}
        end={{ x: 100, y: 0 }}
        arrowColor="#000000"
        circleColor="#FFFFFF"
        iconColor="#FFFFFF"
        type={ArrowType.NonDirectional}
        notes=""
        completed={false}
      />,
```

Reply:
No, not anymore.